### ISSUE TITLE
Automate examples update to latest Cypress version

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/github-action#767]
## Current situation

When a new version of Cypress is released, the [examples](https://github.com/cypress-io/github-action/tree/master/examples) are not automatically updated to use this version.

For examples based on [npm](https://docs.npmjs.com/) technology, [scripts/npm-install-examples.sh](https://github.com/cypress-io/github-action/blob/master/scripts/npm-install-examples.sh) updates Cypress to the latest version. This can be invoked manually through `npm run install-deps`. The settings in [.npmrc](https://github.com/cypress-io/github-action/blob/master/.npmrc) determine that an exact version number is used.

In previous PRs, for instance PR https://github.com/cypress-io/github-action/pull/747, examples based on the [Yarn 1 (Classic) package manager](https://classic.yarnpkg.com/en/) and the [pnpm package manager](https://pnpm.io/) were updated manually and individually:

1. [examples/start-and-yarn-workspaces](https://github.com/cypress-io/github-action/tree/master/examples/start-and-yarn-workspaces) was updated using:
`yarn upgrade-interactive --latest`

2. [examples/install-command](https://github.com/cypress-io/github-action/tree/master/examples/install-command) was updated using:
`yarn add cypress -D`

3. [examples/basic-pnpm](https://github.com/cypress-io/github-action/tree/master/examples/basic-pnpm) was updated using:
`pnpm update --latest`

4. The run command in [.github/workflows/example-install-only.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-install-only.yml) `run: npm i cypress@12.x.x` was manually edited.

*[examples/v9](https://github.com/cypress-io/github-action/tree/master/examples/v9) is frozen to use the last Cypress 9.x version (9.7.0) and is not expected to change.*

## Suggested improvement

Provide better automation for easier updates to Cypress latest version, consistently applied.

See also https://github.com/cypress-io/github-action/pull/747#pullrequestreview-1279647618.

**Comment by MikeMcC399** [comment]
@jaffrepaul 

I have automated updating to Cypress latest in branch [build/update-cypress-latest](https://github.com/MikeMcC399/github-action/tree/build/update-cypress-latest).

The changes are [here](https://github.com/cypress-io/github-action/compare/master...MikeMcC399:github-action:build/update-cypress-latest). I will submit a PR once the backlog of open PRs has been reduced and after I have had the chance to test it a little bit more.

I replaced the original script since it wasn't a DRY (Don't Repeat Yourself) thing and instead I wrote a do-loop to go through the npm examples.
